operators/xr/README.md

```diff
@@ -0,0 +1,275 @@
+
+## Overview
+
+The XR operators provide a complete framework for building Holoscan applications that can render to XR headsets. The implementation leverages OpenXR for cross-platform XR support, Vulkan for graphics, and CUDA for compute operations, enabling high-performance real-time XR applications.
+
+## Core Components
+
+### Frame Management Operators
+
+#### XrBeginFrameOp
+
+- Synchronizes application with XR device timing
+- Calls `xrWaitFrame` and `xrBeginFrame`
+- Outputs frame state for downstream operators
+
+#### XrEndFrameOp
+
+- Submits composition layers to XR device
+- Calls `xrEndFrame` with rendered content
+- Handles multiple composition layers
+
+### XrSession
+
+The central resource that manages the OpenXR session lifecycle:
+
+- **Initialization**: Creates OpenXR instance, system, session, and Vulkan graphics context
+- **Plugin System**: Supports extensible plugins for additional XR functionality
+- **Space Management**: Provides reference and view spaces for pose tracking
+- **Graphics Integration**: Manages Vulkan instance, device, and queue creation
+
+### Rendering Infrastructure
+
+#### XrSwapchainCuda
+
+- Manages OpenXR swapchains with CUDA interop
+- Provides Vulkan/CUDA memory sharing via external memory
+- Supports multiple formats: RGBA8, depth buffers
+- Handles GPU synchronization between CUDA and Vulkan
+
+#### XrCompositionLayerManager
+
+- Manages color and depth swapchains
+- Creates composition layer storage for frame rendering
+- Provides simplified interface for XR rendering
+
+#### XrCompositionLayerProjectionStorage
+
+- Stores projection layer views and depth information
+- Supports stereo rendering with side-by-side layout
+- Handles depth range and near/far plane configuration
+
+### Input Tracking
+
+#### XrHandTracker
+
+- Implements OpenXR hand tracking extension
+- Provides joint locations for both hands
+- Supports 26 hand joints per hand
+- Real-time pose tracking with validation
+
+## Features
+
+### Graphics Pipeline
+
+- **Vulkan Integration**: Native Vulkan graphics with OpenXR
+- **CUDA Interop**: Direct CUDA memory access for compute operations
+- **Multi-format Support**: Color (RGBA8) and depth (D16/D32) formats
+- **Stereo Rendering**: Side-by-side layout for VR headsets
+
+### Input Systems
+
+- **Hand Tracking**: Full hand joint tracking with OpenXR EXT_hand_tracking
+- **Pose Tracking**: 6DOF head and controller tracking
+- **Space Management**: Reference and view space coordinate systems
+
+### Performance Optimizations
+
+- **GPU Synchronization**: Efficient CUDA-Vulkan synchronization
+- **Memory Sharing**: Zero-copy memory between CUDA and Vulkan
+- **Command Buffering**: Pre-recorded Vulkan command buffers
+- **External Semaphores**: Cross-API synchronization primitives
+
+## Dependencies
+
+### External Libraries
+
+- **OpenXR SDK**: 1.0.26 - Cross-platform XR API
+- **OpenXR-Hpp**: 1.0.26 - C++ headers for OpenXR
+- **GLM**: 1.0.1 - Mathematics library for graphics
```

Comment:
Suggest we keep exact versions in `metadata.json` to avoid future mismatch

```suggestion
- **OpenXR SDK**: Cross-platform XR API
- **OpenXR-Hpp**: C++ headers for OpenXR
- **GLM**: Mathematics library for graphics
```